src/cellfinder_core/detect/filters/volume/structure_splitting.py

```diff
@@ -167,20 +171,24 @@ def split_cells(
     if not outlier_keep:
         # TODO: change to checking whether in original cluster shape
         original_max_coords = np.array(original_bounding_cuboid_shape)
-        relative_centres = [
-            x
-            for x in relative_centres
-            if check_centre_in_cuboid(x, original_max_coords)
-        ]
+        relative_centres = np.array(
+            [
+                x
+                for x in relative_centres
+                if check_centre_in_cuboid(x, original_max_coords)
+            ]
+        )
 
-    absolute_centres = []
+    absolute_centres = np.empty((len(relative_centres), 3))
     # FIXME: extract functionality
-    for relative_centre in relative_centres:
-        absolute_centre = Point(
-            orig_corner.x + relative_centre.x,
-            orig_corner.y + relative_centre.y,
-            orig_corner.z + relative_centre.z,
+    for i, relative_centre in enumerate(relative_centres):
+        absolute_centre = np.array(
+            [
+                orig_corner[0] + relative_centre[0],
+                orig_corner[1] + relative_centre[1],
+                orig_corner[2] + relative_centre[2],
+            ]
         )
-        absolute_centres.append(absolute_centre)
+        absolute_centres[i] = absolute_centre
```

Comment:
Might be a bit nicer to write this as the following, which is a bit shorter and avoids allocating a new array.
```python
absolute_centres[:, 0] = orig_corner[0] + relative_centre[0]
absolute_centres[:, 1] = orig_corner[1] + relative_centre[1]
absolute_centres[:, 2] = orig_corner[2] + relative_centre[2]
```